src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1799,6 +1799,9 @@ def _retrieve_segment(
             segments = []
             if single_timestamp_ending:
                 slices.append(len(seek_sequence))
+            else:
+                # we want to include the last timestamp token in the last segment to know it was no single ending
+                slices[-1] += 1
```

Comment:
So you're offsetting the last timestamp by one when the last two tokens are timestamp ? let's say we have `[..., T1, T2]`, you're doing `[..., T1, T2+1]` ?

Reply:
`slices` is a list of indexes along the generated sequence of tokens `seek_sequence`. Let's say we have 100 tokens and it was not single ending, meaning last two tokens are timestamps `[..., T1, T2]`. For this reason, `slices[-1] = 99` yet when we will slice after the segments with `seek_sequence[last_slice:current_slice]`, we want to make sure we include T2 in the slice (so that the we further know it is a double timestamp ending segment as explained in the PR's comment) →  by adding 1 to last slice, we ensure last iteration will slice 'seek_sequence[last_slice:100]` and that T2 will get included